src/lib/OS/SimOS.cc

```diff
@@ -13,8 +13,26 @@ SimOS::SimOS(std::string executablePath,
              std::shared_ptr<simeng::memory::Mem> mem, bool setProcess)
     : executablePath_(executablePath),
       executableArgs_(executableArgs),
+      config_(Config::get()),
       memory_(mem) {
-  syscallHandler_ = std::make_shared<SyscallHandler>(this);
+  // Callback function for communicating a syscall result back to the associated
+  // core
+  auto sendResult = [&](auto result) { sendSyscallResult(result); };
+
+  // Callback function for getting simualted system time
+  auto getTime = [&]() { return getSystemTimer(); };
+
+  // Callback function for translating virtual addresses
+  VAddrTranslator translator = getVAddrTranslator();
+
+  // Callback function for mmaping a file on to host
+  auto mapfd = [&](auto fd, auto len, auto offset) {
+    return mmapHostFd(fd, len, offset);
```

Comment:
for `sendResult`, `getTime` and `mapfd` given there are no variables declared in this function is there a need to capture `&`? Would `[this]` not be the correct scope to capture?

Reply:
Corrected scope